src/orchestrator-process/src/lib.rs

```diff
@@ -525,6 +527,19 @@ impl NamespacedProcessOrchestrator {
                 }
             }
 
+            // Adding scope guard to clean up scratch directory
+            let _guard = scopeguard::guard((), |_| {
+                if let Some(scratch) = scratch_directory {
+                    info!("Cleaning up scratch directory {}", scratch.display());
+                    if let Err(e) = std::fs::remove_dir_all(scratch) {
+                        warn!(
+                            "Error cleaning up scratch directory: {}",
+                            e.display_with_causes()
+                        )
+                    }
+                }
+            });
```

Comment:
This is a potentially slow blocking API in an async context, which will chew up a Tokio worker thread. Better to use [`tokio::fs::remove_dir_all`](`https://docs.rs/tokio/latest/tokio/fs/fn.remove_dir_all.html`) here. Of course, that's asynchronous, so you'll need to do that inside of a `tokio::spawn`. Should be fine though, because all you're doing on error is warning.

Reply:
Makes sense. Let me change that 👍